src/utils/redisCache.js

```diff
@@ -0,0 +1,53 @@
+let crypto = require('crypto')
+let _ = require('lodash/fp')
+let F = require('futil')
+let deterministic_stringify = require('json-stable-stringify')
+
+let redisCache = config => {
+  let { redisCache, redisPrefix = 'es-cache:' } = config
+  let redisClient = F.maybeCall(config.getRedisClient)
+
+  // to resolve racing condition with the same promise
+  let pending = new Map()
+  let dedupedAsync = f => key => {
+    let promise = pending.get(key)
+    if (!promise) {
+      promise = f(key)
+      pending.set(key, promise)
+      promise.finally(() => pending.delete(key))
+    }
+    return promise
+  }
+
+  return (search, schema) => (request, options = {}) => {
+    let { caching: { ttlSecs } = {} } = schema
+
+    if (options.bypassCache || !redisCache || !redisClient || !ttlSecs)
```

Comment:
I don’t think doing this on options quite solved the use case because those are set at the provider level, not request level - but I won’t hold up the merge for this

Reply:
Yeah we would just need contexture to pass this option if there is a flag in tree